sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -67,6 +73,359 @@ describe(
+          firstDisbursementInitialValues: {
+            coeStatus: COEStatus.completed,
+          },
+        },
+      );
+      // Create pre-existing 3 notifications for the above created disbursement.
+      let notificationCounter = 3;
+      while (notificationCounter > 0 && notificationCounter--) {
+        await saveFakeNotification(
+          db.dataSource,
+          {
+            user: student.user,
+            creator: systemUsersService.systemUser,
+            notificationMessage: {
+              id: NotificationMessageType.StudentNotificationDisbursementBlocked,
+            } as NotificationMessage,
+          },
+          {
+            initialValue: {
+              messagePayload: {
+                email_address: student.user.email,
+                template_id:
+                  NotificationMessageType.StudentNotificationDisbursementBlocked,
+                personalisation: {
+                  givenNames: student.user.firstName ?? "",
+                  lastName: student.user.lastName,
+                },
+              },
+              metadata: {
+                disbursementId:
+                  application.currentAssessment.disbursementSchedules[0].id,
+              },
+            },
+          },
+        );
+      }
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processECert(mockedJob.job);
+
+      // Assert
+      expect(result).toStrictEqual([
+        "Process finalized with success.",
+        "Generated file: none",
+        "Uploaded records: 0",
+      ]);
+      const studentNotificationCount = await db.notification.count({
+        where: {
+          notificationMessage: {
+            id: NotificationMessageType.StudentNotificationDisbursementBlocked,
+          },
+          metadata: {
+            disbursementId:
+              application.currentAssessment.disbursementSchedules[0].id,
+          },
+          user: { id: application.student.user.id },
+          creator: { id: systemUsersService.systemUser.id },
+        },
+      });
+      expect(studentNotificationCount).toBe(3);
+    });
+
+    it("Should not create a notification for the student for a disbursement when an attempt is made to create the 2nd notification before 7 days from the first notification.", async () => {
+      // Arrange
+      // Student with invalid SIN to block the disbursement.
+      const student = await saveFakeStudent(db.dataSource, undefined, {
+        sinValidationInitialValue: {
+          isValidSIN: false,
+        },
+      });
+      // Valid MSFAA Number.
+      const msfaaNumber = await db.msfaaNumber.save(
+        createFakeMSFAANumber(
+          { student },
+          {
+            msfaaState: MSFAAStates.Signed,
+            msfaaInitialValues: {
+              offeringIntensity: OfferingIntensity.partTime,
+            },
+          },
+        ),
+      );
```

Comment:
This part is repeating in almost every test, I would suggest to move it to local method.

Reply:
Thanks for the refactor.